### PR TITLE
[MOB-6734] [MOB-6639] Implement placements and resolveMessages()

### PR DESCRIPTION
**Pull request by anlinguist**
## 🔹 Jira Ticket(s)

* [MOB-6734](https://iterable.atlassian.net/browse/MOB-6734)
* [MOB-6639](https://iterable.atlassian.net/browse/MOB-6639)

## ✏️ Description

> This PR implements placements [MOB-6734] and a new helper method to resolve image data [MOB-6639]. Message order is maintained after asynchronously obtaining the image data. Also updates tests to work with new changes.

[MOB-6734]: https://iterable.atlassian.net/browse/MOB-6734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-6639]: https://iterable.atlassian.net/browse/MOB-6639?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-6734]: https://iterable.atlassian.net/browse/MOB-6734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-6639]: https://iterable.atlassian.net/browse/MOB-6639?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/EmbeddedMessagingProcessor.swift [hunk 1]
```diff
@@ -5,51 +5,48 @@
...
-    
-    // MARK: - PRIVATE/INTERNAL
-    
-    private let currentMessages: [IterableEmbeddedMessage]
-    private let fetchedMessages: [IterableEmbeddedMessage]
-    
-    private func getNewMessages() -> [IterableEmbeddedMessage] {
-        let currentMessageIds = currentMessages.map { $0.metadata.messageId }
+    func newlyRetrievedMessages() -> [Int: [IterableEmbeddedMessage]] {
+        var newMessages: [Int: [IterableEmbeddedMessage]] = [:]
```
Comment (evantk91) [inline swift-sdk/Internal/EmbeddedMessagingProcessor.swift:18]: Looks like this is processing embedded messages and storing them in a key pair structure.

**Review by evantk91 — APPROVED** [review]
LGTM. I have a hard time following the image processing. @Ayyanchira can you take a look at that?

File: swift-sdk/Internal/EmptyEmbeddedManager.swift [hunk 2]
```diff
@@ -5,6 +5,9 @@
 import Foundation
 
 class EmptyEmbeddedManager: IterableEmbeddedManagerProtocol {
```
Comment (Ayyanchira) [inline swift-sdk/Internal/EmptyEmbeddedManager.swift:7]: Whats the purpose of EmptyEmbeddedManager. Not sure why we have that for InAppManager as well!

File: swift-sdk/Internal/IterableEmbeddedManager.swift [hunk 3]
```diff
@@ -5,6 +5,29 @@
...
+         message: IterableEmbeddedMessage) {
+        self.title = title
+        self.description = description
+        self.image = image
+        self.buttonText = buttonText
+        self.buttonTwoText = buttonTwoText
+        self.message = message
+    }
+}
+
```
Comment (Ayyanchira) [inline swift-sdk/Internal/IterableEmbeddedManager.swift:30]: Can move to the bottom as it grabs unnecessary attention when looking at EmbeddedManager class.

File: swift-sdk/Internal/IterableEmbeddedManager.swift [hunk 4]
```diff
@@ -101,19 +190,31 @@ class IterableEmbeddedManager: NSObject, IterableEmbeddedManagerProtocol {
...
+        trackedMessageIds = trackedMessageIds.intersection(currentUniqueKeys)
+    }
+
     private func trackNewlyRetrieved(_ processor: EmbeddedMessagingProcessor) {
-        for message in processor.newlyRetrievedMessages() {
-            IterableAPI.track(embeddedMessageReceived: message)
+        for (placementId, messages) in processor.newlyRetrievedMessages() {
+            for message in messages {
+                let messageId = message.metadata.messageId
+                let uniqueKey = "\(placementId)-\(messageId)"
```
Comment (Ayyanchira) [inline swift-sdk/Internal/IterableEmbeddedManager.swift:207]: what is this unique key? 
placementId - messageId? what was the purpose of this?

**Review by evantk91 — APPROVED** [review]
LGTM. Tested on fiterable. Messages are being removed properly.

**Pull request merged**